doc/tutorials/12-constant_pH/12-constant_pH.ipynb

```diff
@@ -210,152 +372,308 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# create the particles\n",
-    "##################################################\n",
     "# we need to define bonds before creating polymers\n",
     "hb = HarmonicBond(k=30, r_0=1.0)\n",
     "system.bonded_inter.add(hb)\n",
     "\n",
-    "# create the polymer composed of ionizable acid groups, initially in the ionized state\n",
+    "# create the polymer positions \n",
     "polymers = polymer.linear_polymer_positions(n_polymers=1,\n",
-    "                                            beads_per_chain=N_acid,\n",
+    "                                            beads_per_chain=N_ACID,\n",
     "                                            bond_length=0.9, seed=23)\n",
-    "for polymer in polymers:\n",
-    "    for index, position in enumerate(polymer):\n",
-    "        id = len(system.part)\n",
-    "        system.part.add(id=id, pos=position, type=TYPE_A, q=q_A)\n",
+    "# add the polymer particles composed of ionizable acid groups, initially in the ionized state\n",
+    "for local_polymer in polymers:\n",
```

Comment:
There is a name conflict with the imported module: 
`from espressomd import electrostatics, polymer, reaction_ensemble`

For consistency, I would either change all imports or change the name of the local iteration variable. 

I'll change the bond creation.

Reply:
it's good practice to use the namespacing of python, *.i.e.* instead of:
``` python
from espressomd import electrostatics
```
you should do:
``` python
import espressomd.electrostatics
```
 that way you can use the module via ```espressomd.electrostatics```, same goes for polymer and reaction ensemble.